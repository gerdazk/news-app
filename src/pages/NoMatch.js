import React from "react";
import HomeButton from "../components/HomeButton";

function NoMatch() {
  return (
    <div className="noMatch">
      <h1 data-testid="noMatch" className="noMatchHeading">
        404: the page does not exist
      </h1>
      <HomeButton text="Go back to the home page" />
    </div>
  );
}

export default NoMatch;
