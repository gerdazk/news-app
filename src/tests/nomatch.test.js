import NoMatch from "../pages/NoMatch";
import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";

describe("No match component: ", () => {
  it("To show text of 404 page", () => {
    const wrapper = mount(
      <Router>
        <NoMatch />
      </Router>
    );
    const data = wrapper.find('[data-testid="noMatch"]').text();
    expect(data).toMatchInlineSnapshot('"404: the page does not exist"');
  });

  it("To match snapshot", () => {
    const wrapper = mount(
      <Router>
        <NoMatch />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
