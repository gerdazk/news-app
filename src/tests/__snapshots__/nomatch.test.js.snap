// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No match component:  To match snapshot 1`] = `ReactWrapper {}`;
